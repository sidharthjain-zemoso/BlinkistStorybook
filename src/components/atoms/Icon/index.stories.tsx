import { ComponentStory, ComponentMeta } from "@storybook/react";

import Icon from "./index";

export default {
  title: "Atoms/Icon",
  component: Icon,
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} />;

export const ClockIcon: ComponentStory<typeof Icon> = Template.bind({});
ClockIcon.args = {
  url: "/res/icons/clock.png",
  sx: {},
};
