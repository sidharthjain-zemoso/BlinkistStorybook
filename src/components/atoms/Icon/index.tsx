type IconProps = {
  url: string;
  sx?: Object;
  className?: string;
};

const Icon: React.FC<IconProps> = (props) => {
  return (
    <img
      className={props.className}
      src={process.env.PUBLIC_URL + props.url}
      style={{ marginRight: 5, ...props.sx }}
      alt="icon"
    />
  );
};

export default Icon;
