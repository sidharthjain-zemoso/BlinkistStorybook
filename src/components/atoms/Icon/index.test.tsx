import { render, screen } from "@testing-library/react";
import Icon from "./";

test("renders icon component", () => {
  render(<Icon url="res/icons/up-arrow.png" />);

  const imageElement = screen.getByRole("img");
  expect(imageElement).toMatchSnapshot();
});
