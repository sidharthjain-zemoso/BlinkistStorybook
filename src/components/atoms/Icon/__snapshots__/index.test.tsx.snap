// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders icon component 1`] = `
<img
  alt="icon"
  src="res/icons/up-arrow.png"
  style="margin-right: 5px;"
/>
`;
